HopsanGenerator/src/generators/HopsanFMIGenerator.cpp

```diff
@@ -1310,6 +1310,31 @@ bool HopsanFMIGenerator::generateToFmu(QString savePath, ComponentSystem *pSyste
     fmuHopsanSourceCode.replace("<<<setdataptrs>>>", setDataPtrsString);
     fmuHopsanSourceCode.replace("<<<timestep>>>", QString::number(pSystem->getDesiredTimeStep()));
 
+    QList<ParameterSpecification> parSpecs;
+    getParameters(parSpecs, pSystem);
+    QString addParametersToMap;
+    for(const auto &parSpec : parSpecs) {
+        QString dataType;
+        if(parSpec.type == "Real") {
+            addParametersToMap.append(QString("realParametersMap.insert(std::pair<int,hopsan::HString>(%1,\"%2\"));").arg(vr).arg(parSpec.name.toStdString().c_str()));
+        }
+        else if(parSpec.type == "Integer") {
+            addParametersToMap.append(QString("intParametersMap.insert(std::pair<int,std::string>(%1,\"%2\"));").arg(vr).arg(parSpec.name));
+        }
+        else if(parSpec.type == "Boolean") {
+            addParametersToMap.append(QString("boolParametersMap.insert(std::pair<int,std::string>(%1,\"%2\"));").arg(vr).arg(parSpec.name));
+        }
+        else if(parSpec.type == "String") {
+            addParametersToMap.append(QString("stringParametersMap.insert(std::pair<int,std::string>(%1,\"%2\"));").arg(vr).arg(parSpec.name));
+        }
+        else {
+            continue;   //Illegal data type, should never happen
+        }
+        QString temp = QString("dataPtrs[%1] = static_cast<%2*>(spCoreComponentSystem->getParameterDataPtr(\"%3\"));").arg(vr).arg(dataType).arg(parSpec.name);
```

Comment:
Has data type been set here ? Seems like it has not been assign

Reply:
Also I looked for this in the generated code, but it is not there at all ?So this entire line is completely discarded it seems like.

Reply:
It was not used, since parameters do not use data pointers. I removed the unused code.